src/System.Net.Security/src/System/Net/Security/SniHelper.cs

```diff
@@ -0,0 +1,278 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System.Net.Security
+{
+    internal class SniHelper
+    {
+        private static IdnMapping s_idnMapping = CreateIdnMapping();
+
+        public static string GetServerName(byte[] clientHello)
+        {
+            return GetSniFromSslPlainText(clientHello);
+        }
+
+        // https://tools.ietf.org/html/rfc6101#section-5.2.1
+        // SSLPlainText structure:
+        //   - ContentType (1 byte) => 0x16 is handshake
+        //   - ProtocolVersion version (2 bytes)
+        //   - uint16 length
+        //   - opaque fragment[SSLPlaintext.length]
```

Comment:
I usually expect comments above a method to be about calling it.  And comments within a method to be justifying the magic numbers.